src/layout/Header.jsx

```diff
@@ -0,0 +1,167 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import md5 from "md5";
+
+import NavbarMobile from "../components/NavbarMobile";
+import { NavLink } from "react-router-dom";
+
+export default function Header() {
+  const user = useSelector((store) => store.client.user);
+  const categories = useSelector((store) => store.product.categories);
+
+  const gravatarUrl = user.email
```

Comment:
gravatarUrl'i useMemo ile cache'le, her render'da hesaplanmasın

Reply:
gravatarUrl için user.email kontrolü iyi, ama user nesnesinin varlığını da kontrol et. Aksi halde, user tanımsız olduğunda hata alabilirsin